helpful_tools/Install-DockerCE/install-docker-ce.ps1

```diff
@@ -571,10 +576,49 @@ Install-Docker()
 
     Test-Admin
 
-    Write-Output "Installing Docker..."
+    try {
+        #If one of these are set to default then the whole .zip needs to be downloaded anyways.
+        Write-Output "DOCKER $DockerPath"
+        if ($DockerPath -eq "default" -or $DockerDPath -eq "default") {
+            Write-Output "Checking Docker version"
+            $version = ((Invoke-WebRequest -Uri $DefaultDockerLocation).Links | Where-Object {$_.href -like "docker*"}).href | Sort-Object -Descending | Select-Object -First 1  | Select-String -Pattern "docker-(\d+\.\d+\.\d+).+"  -AllMatches | Select-Object -Expand Matches | %{ $_.Groups[1].Value }
+
+            if($DockerVersion -ne "latest") {
+                $version = $DockerVersion
+            } else {
+                if(!(((Invoke-WebRequest -Uri $global:DefaultDockerLocation).Links | Where-Object {$_.href -like "*$global:DockerVersion*"}).href | Sort-Object -Descending | Select-Object -First 1)) {
+                    Write-Error "Docker version supplied $global:DockerVersion was invalid, using the latest version."
```

Comment:
It feels like these line should be less dense/not one-liners, using multiple assignments to increase readability. Falling back to latest when getting an invalid version also seems somewhat unexpected.

Reply:
Good point, I just updated the logic. Any better?